source/common/http/conn_pool_grid.h

```diff
@@ -43,17 +43,23 @@ class ConnectivityGrid : public ConnectionPool::Instance,
   // It also relays cancellation calls between the original caller and the
   // current connection attempts.
   class WrapperCallbacks : public ConnectionPool::Cancellable,
-                           public LinkedObject<WrapperCallbacks> {
+                           public LinkedObject<WrapperCallbacks>,
+                           public Event::DeferredDeletable {
   public:
     WrapperCallbacks(ConnectivityGrid& grid, Http::ResponseDecoder& decoder,
                      ConnectionPool::Callbacks& callbacks, const Instance::StreamOptions& options);
 
+    bool hasNotifiedCaller() { return inner_callbacks_ == nullptr; }
+    void deleteIsPending() override { next_attempt_timer_.reset(); }
```

Comment:
What's the relationship between the next attempt timer and delete pending?

Reply:
commented inline

Reply:
Oh! I see. I missed that this was a virtual method as part of `Event::DeferredDeletable`. Mind adding:
``// Event::DeferredDeletable:`

Reply:
done